src/CoreWCF.Primitives/src/CoreWCF/Configuration/ServiceBuilderExtensions.cs

```diff
@@ -15,5 +15,23 @@ public static class ServiceBuilderExtensions
                 .GetRequiredService<ServiceConfigurationDelegateHolder<TService>>();
             holder.AddConfigDelegate(func);
         }
+
+        public static void ConfigureServiceHostBase(this IServiceBuilder builder, Type serviceType, Action<ServiceHostBase> func)
+        {
+            if (serviceType is null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException(nameof(serviceType)));
+            }
+
+            if (!serviceType.IsClass)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException($"{serviceType.FullName} should be a class", nameof(serviceType)));
```

Comment:
Can you add exception text to strings.resx and use SR.Format(SR.ResourceName, serviceType.FullName)

Reply:
sure!